src/lib/FlightTasks/tasks/AutoMapper/FlightTaskAutoMapper.hpp

```diff
@@ -77,6 +77,10 @@ class FlightTaskAutoMapper : public FlightTaskAuto
 private:
 
 	void _reset(); /**< Resets member variables to current vehicle state */
+	void _checkPositionLock(); /**< check whether position should be locked */
 	WaypointType _type_previous{WaypointType::idle}; /**< Previous type of current target triplet. */
 	bool _highEnoughForLandingGear(); /**< Checks if gears can be lowered. */
+	bool _position_locked{false};
+	bool _request_position_lock{false};
+	matrix::Vector3f _locked_position; /**< position at which vehicle is locked */
```

Comment:
You don't have to make an extra state for the locked position but can just set the setpoint and make sure it doesn't get overwritten. If you can't find out if it is overwritten somewhere (which I wouldn't blame you for) the task is too complicated and we have to fix it.